tests/60app-services/05lookup3pe.pl

```diff
@@ -0,0 +1,117 @@
+test "HS will proxy request for 3PU mapping",
+   requires => [ local_user_fixture(), $main::APPSERV[0] ],
+
+   do => sub {
+      my ( $user, $appserv ) = @_;
+
+      Future->needs_all(
+         $appserv->await_http_request( "/3pu/protocol", sub { 1 } )->then( sub {
+            my ( $request ) = @_;
+
+            assert_deeply_eq(
+               { $request->query_form },
+               {
+                  field1 => "ONE",
+                  field2 => "TWO",
+               },
+               'fields in received AS request'
+            );
+
+            $request->respond_json( [
+               {
+                  protocol => "protocol",
+                  fields   => { field1 => "result" },
+                  userid   => '@remote-user:bridged.example.com',
+               }
+            ] );
+
+            Future->done(1);
+         }),
+
+         do_request_json_for( $user,
+            method => "GET",
+            uri    => "/unstable/3pu/protocol",
```

Comment:
If the protocol is an arbitrary protocol for testing can it be something less sensible?